update guide with traffic policy (#737)
ngrok/ngrok-docs: docs/k8s/user-guide.mdx

                  number: <service-port>
```

## Traffic Policy

> Desigining and applying just traffic policies to inbound and outbound traffic is a simplified best practice for handling edge management.

Reviewer: ```suggestion
Desigining and applying just traffic policies to inbound and outbound traffic is a simplified best practice for handling edge management.
```
```suggestion
Traffic policies can be applied to inbound and and outbound traffic for simplified edge management.
```

> Desigining and applying just traffic policies to inbound and outbound traffic is a simplified best practice for handling edge management.

Reviewer: ```suggestion
Traffic policies for inbound and outbound traffic can simplify edge management.
```

### Design

> Defining a traffic policy resource with rules composed of expressions and actions allows you to influence and control traffic via the policy engine.

Reviewer: This language might be a little clearer.
```suggestion
`NgrokTrafficPolicy` objects can be defined with rules composed of expressions and actions that validate and filter traffic via the policy engine.
```

Author: This was tricky, I got caught up in the `influx and control traffic` phrasing for traffic policies in `docs/http/index.mdx` and the three other `docs/{http,tcp,tls}/traffic-policy/index.mdx`, but this nails it.

```yaml
kind: NgrokTrafficPolicy
apiVersion: ngrok.k8s.ngrok.com/v1alpha1
metadata:
  name: traffic-policy-example
spec:
  policy:
    inbound:
      - name: "greetings"
        expressions:
          - "req.Method == 'GET'"
          - "req.Path.startsWith('/greetings')"
        actions:
          - type: "custom_response"
            config:
              status_code: 200
              content: Hello
              headers:
                content-type: text/plain
```

Learn more about the policy expressions and actions here:

- [Policy Expressions](https://ngrok.com/docs/http/traffic-policy/expressions/)
- [Policy Actions](https://ngrok.com/docs/http/traffic-policy/actions/)

### Composition

> With a traffic policy designed for your production ingress use case you can easily add the policy and deploy with an annnotation.

Reviewer: ```suggestion
Traffic policies can be added to Ingress objects using annotations.
```

```yaml
---
kind: NgrokTrafficPolicy
apiVersion: ngrok.k8s.ngrok.com/v1alpha1
metadata:
  name: traffic-policy-test
spec:
  policy:
    inbound:
      - name: "deny_patch"
        expressions:
          - "req.Method == 'PATCH'"
        actions:
          - type: "deny"
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: example
  annotations:
    k8s.ngrok.com/traffic-policy: traffic-policy-test
spec:
  ingressClassName: ngrok
  rules:
    - host: <my-host>.ngrok.app
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: <service-name>
                port:
                  number: <service-port>
```

> With just the `k8s.ngrok.com/traffic-policy` annotation we've applied our policy to ingress.

Reviewer: I'm not sure this line adds much.
```suggestion
```

:::note

> Utilize either the module set or the traffic policy annotations on an ingress. Trying to utilize both features will result in an error.

Reviewer: ```suggestion
Using both an `NgrokModuleSet` and an `NgrokTrafficPolicy` will result in an error.
```
